zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationMigrateProcessor.java

```diff
@@ -274,10 +285,61 @@ private void tryMigrateElementInstance(
                         .setBpmnProcessId(targetProcessDefinition.getBpmnProcessId())
                         .setTenantId(elementInstance.getValue().getTenantId())));
 
+    final var context = new BpmnElementContextImpl();
+    context.init(elementInstance.getKey(), elementInstanceRecord, elementInstance.getState());
+    final var targetElement =
+        targetProcessDefinition
+            .getProcess()
+            .getElementById(targetElementId, ExecutableActivity.class);
+    final List<DirectBuffer> subscribedMessageNames = new ArrayList<>();
+    catchEventBehavior
+        .subscribeToEvents(
```

Comment:
🤔 Such a comment could potentially go out of sync because it may be forgotten when the conditions change. 

❓ I described it in the commit message of bfff24d9f906482e428c63b7b8a5523925ee64af. Do you think we should still add a comment, though?